certora/specs/SocialRecoveryModule.spec

```diff
@@ -466,3 +466,30 @@ rule invalidatingNonceInRecovery(env e, address guardian, address[] newOwners, u
     bool isReverted = lastReverted;
     assert success => isReverted;
 }
+
+// This rule verifies that the safe can only add/revoke guardians and thresholds for itself, and not for other safe contracts.
+rule safeContractAccessToGuardiansAndThreshold(env e, address guardian, uint256 threshold, address otherSafeContract) {
+    requireGuardiansLinkedListIntegrity(guardian);
+    require e.msg.sender == safeContract;
+    require safeContract != otherSafeContract;
+
+    bool isGuardianInOtherSafeContract = currentContract.isGuardian(otherSafeContract, guardian);
+
+    currentContract.addGuardianWithThreshold@withrevert(e, guardian, threshold);
+    bool addGuardianSuccess = !lastReverted;
```

Comment:
could you please explain why is the check for success needed here? doesn't it only consider non-reverting paths if you omit `withrevert`?

Reply:
You are right, updated the rule.